templates/detail.jinja2

```diff
@@ -1,38 +1,86 @@
-<!DOCTYPE html>
-<html>
-<head>
-  <meta charset="utf-8">
-  <title>Tyler Peek | Detail</title>
-  <link href="/static/style.css" rel="stylesheet">
-  <link href="/static/codehilite.css" rel="stylesheet">
-  <link href='http://fonts.googleapis.com/css?family=Homemade+Apple' rel='stylesheet' type='text/css'>
-</head>
+{% extends "base.jinja2" %}
+{% block section %}
+<nav>
+  <ul>
+    <li><a href="{{ request.route_url('home') }}">Home</a></li>
+    {% if request.authenticated_userid %}
+    <li><a id="add" href="{{ request.route_url('add') }}">Add</a></li>
+    <li><a id="edit" href="{{ request.route_url('edit', entry_id=id) }}">Edit</a></li>
+    <li><a id="logout" href="{{ request.route_url('logout') }}">Log Out</a></li>
+    {%- endif %}
+  </ul>
+</nav>
+</header>
 <body>
-  <header>
-  <a href="{{ request.route_url('home') }}" id="logo">
-      <h1 class="name">Tyler Peek</h1>
-      <h2>Python Developer</h2>
-    </a>
-    <nav>
-      <ul>
-        <li><a href="{{ request.route_url('home') }}">Home</a></li>
-        {% if request.authenticated_userid %}
-        <li><a id="add" href="{{ request.route_url('add') }}">Add</a></li>
-        <li><a id="edit" href="{{ request.route_url('edit', entry_id=id) }}">Edit</a></li>
-        <li><a id="logout" href="{{ request.route_url('logout') }}">Log Out</a></li>
-        {%- endif %}
-      </ul>
-    </nav>
-  </header>
+  <div id="fb-root"></div>
   <div id="wrapper">
-    <section class="highlight">
-      <h3 class="title">{{ title }} ---- {{ time }}</h3>
-      <p class="text">{{ text|safe }}</p>
+    <div id="myDiv"></div>
+    {% if request.authenticated_userid %}
+    <button id="edit_entry" type="button" onclick="loadXMLDoc()">Edit Entry</button>
+    {% endif %}
+    <section id="entry" class="highlight">
+      <h3 id="title1" class="title">{{ time }} ---- {{ title }}</h3>
+      <div id="text1" class="text">{{ text|safe }}</div>
     </section>
-    <footer>
-    </a>
-    <p>&copy; 2015 Tyler Peek.</p>
-  </footer>
-</div>
+    <div><a href="https://twitter.com/share" class="twitter-share-button" data-via="TylerJPeek" data-size="large">Tweet</a></div><br>
+    <div class="fb-share-button" data-href="https://developers.facebook.com/docs/plugins/" data-layout="button_count"></div>
+  </div>
 </body>
-</html>
+
+<script>
+  document.getElementById('label').innerHTML = '{{ title }}';
+  function loadXMLDoc() {
+    document.getElementById('myDiv').style.display = 'initial';
+    document.getElementById('edit_entry').style.display = 'none';
+    document.getElementById('entry').style.display = 'none';
+    xmlhttp = new XMLHttpRequest();
+    xmlhttp.open("GET","{{ request.route_url('ajax_edit', entry_id=id) }}",true);
+    xmlhttp.send();
+
+    xmlhttp.onreadystatechange = function () {
+      if (xmlhttp.readyState === 4) {
+        document.getElementById('myDiv').innerHTML = xmlhttp.responseText;
+      }
+    }
+  }
```

Comment:
yikes.  Why not just use the jQuery `$.ajax()` method?  It's a lot cleaner and more cross-browser safe than old-school manual XMLHttpRequest objects.  I haven't seen code like this since jQuery was introduced.